src/full/Agda/TypeChecking/Serialise/Instances/Internal.hs

```diff
@@ -433,46 +435,46 @@ instance EmbPrj Defn where
     -- Andreas, 2024-10-27
     -- DataOrRecSig is possible via unquoteDecl in meta-programming, see #7576
 
-  value = vcase valu where
-    valu [0, a]                                        = valuN Axiom a
-    valu [1, a, b, s, u, c, d, e, f, g, h, i, j, k]
-                                                       = valuN (\ a b s -> Function a b s Nothing) a b s u c d e f g h i j k
-    valu [2, a, b, c, d, e, f, g, h, i, j]             = valuN Datatype a b c d e f g h i j
-    valu [3, a, b, c, d, e, f, g, h, i, j, k, l, m]    = valuN Record   a b c d e f g h i j k l m
-    valu [4, a, b, c, d, e, f, g, h, i, j, k]          = valuN Constructor a b c d e f g h i j k
-    valu [5, a, b, c, d, e, f]                         = valuN Primitive   a b c d e f
-    valu [6, a, b]                                     = valuN PrimitiveSort a b
-    valu [7, a]                                        = valuN GeneralizableVar a
-    valu [8, a]                                        = valuN DataOrRecSig a
-    valu _                                             = malformed
+  value = vcase \case
+    N2 0 a                                   -> valuN Axiom a
+    N6 1 a b s u c (N6 d e f g h i (N2 j k)) -> valuN (\ a b s -> Function a b s Nothing)
```

Comment:
It is a pity that Haskell has less magic on the lhs than on the rhs, so we can't have a smart pattern synonym `N` that does not require us to manually chunk the 14 arguments into 6 + 6 + 2, or can we?